### PR TITLE
Issue #45: add version number and invocation parameters to log

### PR DESCRIPTION
**Pull request by ensembledeanw**

### CODE REVIEW

File: user_sync/app.py [hunk 1]
```diff
@@ -299,6 +299,17 @@ def create_config_loader_options(args):
     
     return config_options
 
+def log_parameters(args):
+    '''
+    Log the invocation parameters to make it easier to diagnose problem when customers
```
Comment (adobeDan) [inline user_sync/app.py]: Looks like you lost some text here. "to diagnose problem when customers"?

File: user_sync/app.py [hunk 2]
```diff
@@ -299,6 +299,17 @@ def create_config_loader_options(args):
...
+    :param args: namespace
+    :return: None
+    '''
+    logger.info('----- Invocation Parameters -----')
+    for parameter_name, parameter_value in args.__dict__.iteritems():
+        logger.info('  %s: %s', parameter_name, parameter_value)
+    logger.info('---------------------------------')
```
Comment (adobeDan) [inline user_sync/app.py]: So this looks like it logs our internal (code-level) parameter names and values, which is good, but it doesn't get at the essence of the ask from a customer perspective, which is to actually put the command line that the customer gave us in the log.  Can you please *add* to what you have here additional code which essentially logs the sys.argv command line?

File: user_sync/app.py [hunk 3]
```diff
@@ -309,9 +320,11 @@ def main():
         
         config_loader = create_config_loader(args)
         init_log(config_loader.get_logging_config())
-        
-        run_stats = user_sync.helper.JobStats("Run", divider = "=")
+
+        # add start divider, app version number, and invocation parameters to log
+        run_stats = user_sync.helper.JobStats('Run (Ver: ' + APP_VERSION + ')', divider='=')
```
Comment (adobeDan) [inline user_sync/app.py]: How about "User Sync version:"  No need to be terse, and much clearer for non-English speakers then "Ver".

**Comment by ensembledeanw** [comment]
Thank you for your comments, Dan. I made the changes base on your suggestion and it is ready for review again.

**Pull request merged**